### PR TITLE
cleanup: deleted old docs

### PR DESCRIPTION
**Pull request by droot**
Deleted outdated docs.

### CODE REVIEW

File: docs/declaring_rbac_rules_for_controllers.md [hunk 1]
```diff
@@ -1,30 +0,0 @@
-# Declaring RBAC rules for controllers
```
Comment (mengqiy) [inline docs/declaring_rbac_rules_for_controllers.md]: The content in this file seems to still be valid.
Do we have a replacement somewhere else?

Comment (droot) [reply docs/declaring_rbac_rules_for_controllers.md]: Good catch. Part of the documentation is valid in this file. Filed an issue for 
https://github.com/kubernetes-sigs/kubebuilder/issues/426 for adding new section in kubebuilder book.

Comment (mengqiy) [reply docs/declaring_rbac_rules_for_controllers.md]: I'm fine with doing it in a future PR :)

**Pull request merged**